pyiceberg/table/__init__.py

```diff
@@ -158,9 +158,11 @@
 
 def _check_schema_compatible(table_schema: Schema, other_schema: "pa.Schema") -> None:
     """
-    Check if the `table_schema` is compatible with `other_schema`.
+    Check if the `table_schema` is compatible with `other_schema` in terms of the Iceberg Schema representation.
 
-    Two schemas are considered compatible when they are equal in terms of the Iceberg Schema type.
+    The schemas are compatible if:
+    - All fields in `other_schema` are present in `table_schema`. (other_schema <= table_schema)
+    - All required fields in `table_schema` are present in `other_schema`.
```

Comment:
Just a heads up, with V3 this changes since it is allowed to add required fields with a default value.